format_precice_config/format_precice_config.py

```diff
@@ -117,23 +118,192 @@ def printChildren(self, element, level):
+        sorted_children = sorted(element.getchildren(), key=custom_sort_key)
+
+        last = len(sorted_children)
+        for i, group in enumerate(sorted_children, start=1):
+            # Special handling for participants to reorder child elements
+            if 'participant' in str(group.tag):
+                # Define order for participant child elements with more generalized matching
+                participant_order = {
+                    'provide-mesh': 1,
+                    'receive-mesh': 2,
+                    'write-data': 3,
+                    'read-data': 4,
+                    'mapping:': 5  # Matches mapping:nearest-neighbor, mapping:rbf, etc.
+                }
+                
+                # Sort participant's children based on the defined order
+                sorted_participant_children = sorted(
+                    group.getchildren(), 
+                    key=lambda child: next(
+                        (rank for prefix, rank in participant_order.items() 
+                         if str(child.tag).startswith(prefix)), 
+                        6  # Unknown elements appear last
+                    )
+                )
+                
+                # Separate different types of elements
+                mesh_elements = []
+                data_elements = []
+                mapping_elements = []
+                
+                for child in sorted_participant_children:
+                    if str(child.tag) in ['provide-mesh', 'receive-mesh']:
+                        mesh_elements.append(child)
+                    elif str(child.tag) in ['write-data', 'read-data']:
+                        data_elements.append(child)
+                    elif str(child.tag).startswith('mapping:'):
+                        mapping_elements.append(child)
+                
+                # Construct participant tag with attributes
+                participant_tag = "<{}".format(group.tag)
+                for attr, value in group.items():
+                    participant_tag += ' {}="{}"'.format(attr, value)
+                participant_tag += ">"
+                
+                # Print participant opening tag
+                self.print(self.indent * level + participant_tag)
+                
+                # Print mesh elements
+                for child in mesh_elements:
+                    self.printElement(child, level + 1)
+                
+                # Add newline between mesh and data
+                if mesh_elements and data_elements:
+                    self.print()
+                
+                # Print data elements
+                for child in data_elements:
+                    self.printElement(child, level + 1)
+                
+                # Add newline before mapping
+                if data_elements and mapping_elements:
+                    self.print()
+                
+                # Print mapping elements with multi-line formatting
```

Comment:
Have you tested the mapping formatting with rbf mappings?
They have an additional subtag.

https://precice.org/configuration-xml-reference.html#mappingrbf